### DISCUSSION TITLE
getting trophy title for ps3 and vita

### DISCUSSION THREAD

**Discussion by glebmiller** [isFakeAccount/psnawp#44] [Q&A] [answered]
hello.
how do I get trophy title infos for ps3 and vita games? 
for ps4 and ps5 I use client.trophy_titles_for_title(title_ids=[title_id]) where I get id from  https://andshrew.github.io/PlayStation-Titles/

but there are no ps3 or vita title ids. all I have is np_comminication_id, how can I get prophy info from that?

**Comment by isFakeAccount** [comment]
You could use the `get_title_id` from the search Class. The search class works like the search function on the app. If you can search the ps3 and ps vita games on the mobile app then the search class should work as well. 

I could not since my account is only linked to a PS4.

**Reply by glebmiller** [reply]
unfortunately, get_title_id gets wrong ids. my bot account is in US, and my friends are all from ru, and the title_ids don't match, I have to use Andrew's GitHub title list, but there are no ps3 or vita games

**Reply by isFakeAccount** [reply]
I created an issue for it. Right now I am not quite sure how to fix this issue. I might have to talk to other api wrappers devs

**Comment by isFakeAccount** [comment, marked as answer]
You can use this code example to get all your PS3/PSVITA Trophies, Trophy titles, groups, etc... 

```py
from os import getenv

from dotenv import load_dotenv
from psnawp_api import PSNAWP
from psnawp_api.models.trophies import PlatformType

load_dotenv()


def main() -> None:
    psn = PSNAWP(getenv("NPSSO_CODE", "NPSSO_CODE"))
    client = psn.me()

    print("Trophy Titles")
    trophy_titles = client.trophy_titles(limit=100)
    for trophy_title in trophy_titles:
        if PlatformType.PS3 in trophy_title.title_platform:
            print(trophy_title.title_name, trophy_title.title_platform, trophy_title.earned_trophies, trophy_title.np_communication_id)

    print("\nTrophies")
    for trophy in client.trophies("NPWR05167_00", PlatformType.PS3, limit=10):
        print(trophy.trophy_name)

    print("\nTrophy Group Summary")
    the_lego_movie_tgs = client.trophy_groups_summary("NPWR05167_00", platform=PlatformType.PS3, include_progress=True)
    print(the_lego_movie_tgs)


if __name__ == "__main__":
    main()
```

The main issue is that you cannot search for PlayStation games using Search class. Since you can't find title ids using search function, you can't use `client.trophy_titles_for_title` because it requires title ids. 

But remaining methods and endpoints work fine.
